src/hashcash.cr

```diff
@@ -52,57 +67,42 @@ module Hashcash
     end
 
     def self.parse(stamp : String)
-      # version =
-      # bits =
-      # resource =
-
-      # parse date
-      # (@version, @bits, @date, @resource, ext, rand, counter) = stamp_string.split(':')
-      # puts @version
-      # puts rand
-      # puts counter
+      parts = stamp.split(":")
+      version, bits, date, resource, ext = parts
+      Hashcash::Stamp.new(
+        resource,
+        version.to_i,
+        bits.to_i,
+        Time.parse_utc(date, "%y%m%d%H%M%S"),
+        ext,
+        stamp
+      )
     end
 
-    def verify_stamp(stamp : String, expiry = Time::Span.new(days: 2), bits = 20)
-      split_stamp = stamp.split(":")
-      date = parse_date(split_stamp[2])
+    def verify_stamp(stamp : String, expiry = 2.days, bits = 20)
```

Comment:
To keep methods small, try splitting this up for the checks within the stamp class.

`#valid?(bits = 20) : Bool`

`#expired?(window = 2.days.ago..2.days.from_now) : Bool`

`#is_for?(resource : String) : Bool`

Back on the outer module, I'd recommend then having methods that use these:

`.valid?(stamp : String, resource : String, time_window = 2.day.ago..2.days.from_now, bits = 20) : Bool`

`.verify!(stamp : String, resource: String, time_window = 2.day.ago..2.days.from_now, bits = 20) : Nil` (raises on error)

Reply:
okay cool, will do this